### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by thomasplevy**

### CODE REVIEW

**Comment by thomasplevy** [comment]
Tests are failing because tests rely on changes to the LifterLMS core which aren't released yet

File: includes/class-llms-rest-authentication.php [hunk 1]
```diff
@@ -17,13 +17,271 @@
...
+	 *
+	 * @link https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate
+	 *
+	 * @param WP_REST_Response $response Current response being served.
+	 * @return WP_REST_Response
+	 */
+	public function send_unauthorized_headers( $response ) {
+
+		if ( is_wp_error( $this->get_error() ) ) {
+			$auth_message = __( 'LifterLMS REST API', 'lifterlms' );
```
Comment (eri-trabiccolo) [inline includes/class-llms-rest-authentication.php]: Maybe we want to send a more meaningful error message

Comment (thomasplevy) [reply includes/class-llms-rest-authentication.php]: Check out the link there (https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/WWW-Authenticate).

The WooCommerce rest api does add an error here however from my understanding of the header's intention it's supposed to just describe the area. The error message itself (which would be included in the rest body) would be the error message.

This is just an additional header.

Semantically I think the current message here is correct and boy do I love me some semantics :-p

File: includes/class-llms-rest-authentication.php [hunk 2]
```diff
@@ -17,13 +17,271 @@
...
+		if ( ! empty( $error ) ) {
+			return $error;
+		}
+
+		return $this->get_error();
+
+	}
+
+	/**
+	 * Check if the API Key can preform the request.
```
[the remark this answers is not in the document]
Comment (thomasplevy) [reply includes/class-llms-rest-authentication.php]: Opened issue #36 to track this as I'm not going to do it now and I don't want to forget but I do want to merge

File: includes/class-llms-rest-authentication.php (hunk 1 above)
Comment (eri-trabiccolo) [reply includes/class-llms-rest-authentication.php]: Yes Woo misled me.
I think you're right and I should've checked the @link before questioning.
My bad.

**Pull request merged**